### PR TITLE
Add x no filter property

### PR DESCRIPTION
**Pull request by JSReds**
depends-on: https://github.com/balena-io-modules/rendition/pull/1733/files

### CODE REVIEW

File: src/AutoUI/Filters/Filters.tsx [hunk 1]
```diff
@@ -23,6 +27,100 @@ interface FiltersProps<T> {
...
+		.filter(
+			(propKey) => !Array.isArray(xNoFilter) || !xNoFilter.includes(propKey),
+		)
+		.reduce(
+			(subAcc, propKey) => ({
+				...subAcc,
+				[propKey]: properties[propKey],
+			}),
+			{},
+		);
```
Comment (thgreasi) [inline src/AutoUI/Filters/Filters.tsx:47]:
```suggestion
	if (!properties || !Array.isArray(xNoFilter)) {
		return properties;
	}
	const xNoFilterSet = new Set(xNoFilter);
	return Object.fromEntries(
		Object.keys(properties)
		.filter(
			(propKey) => !xNoFilterSet.has(propKey),
		)
		.map((propKey) => [propKey, properties[propKey]])
	);
```

File: src/oData/jsonToOData.ts [hunk 2]
```diff
@@ -215,18 +216,29 @@ export const convertToPineClientFilter = (
...
 		return null;
 	}
+
 	const { field, reverse, refScheme } = sortInfo;
 	if (!field) {
 		return null;
 	}
 	const direction = !reverse ? 'asc' : 'desc';
+	const customOrderByKey =
+		customSort?.[sortInfo.field as keyof typeof customSort];
```
Comment (thgreasi) [inline src/oData/jsonToOData.ts:233]: NIT:
```suggestion
		customSort?.[field as keyof typeof customSort];
```

File: src/oData/jsonToOData.ts [hunk 3]
```diff
@@ -215,18 +216,29 @@ export const convertToPineClientFilter = (
...
+	if (typeof customOrderByKey === 'string') {
```
Comment (thgreasi) [inline src/oData/jsonToOData.ts:234]: Should we throw if this is not string | null?
Eg if the result is a function, as per `customSort?: Dictionary<(a: T, b: T) => number> | Dictionary<string>;`

File: src/AutoUI/index.tsx [hunk 4]
```diff
@@ -721,7 +736,12 @@ const getColumnsFromSchema = <T extends AutoUIBaseResource<T>>({
 				priority,
 				type: 'predefined',
 				refScheme: refScheme?.[0],
-				sortable: customSort?.[key] ?? getSortingFunction(key, val),
+				sortable:
+					(xNoSort ? false : null) ??
+					(typeof customSort?.[key] !== 'string'
+						? (customSort?.[key] as TableSortFunction<T>)
+						: null) ??
+					getSortingFunction(key, val),
```
Comment (thgreasi) [inline src/AutoUI/index.tsx:744]: Should also check to make sure nobody is passing functions for server side pagination mode or strings for in-memory mode

File: src/AutoUI/Filters/Filters.tsx (hunk 1 above)
Comment (thgreasi) [reply src/AutoUI/Filters/Filters.tsx:47]: TBH, since we already always passing a new object, this could even be a for-of w/ `delete` statements, saving 2 object creations, but no strong opinion on this.

File: src/AutoUI/index.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (JSReds) [reply src/AutoUI/index.tsx]: perfect sense

Comment (JSReds) [reply src/AutoUI/index.tsx]: 👍

File: src/AutoUI/index.tsx
Comment (myarmolinsky) [inline src/AutoUI/index.tsx:1]: Lots of changes in this file so a bit hard to parse: it's mostly reformat, right? What's the main thing to note here?

Comment (JSReds) [reply src/AutoUI/index.tsx:1]: mainly the following:
- removeFieldsWithNoFilter function which remove the filter from the schema if we add x-no-filter property in the description
- orderbyBuilder function where we add support for a customSort 
- inside existing getColumnsFromSchema function (line 646 src/AutoUI/index.tsx) where we add the logic to remove the sorting if porperty `x-no-sort` is enabled

**Review by myarmolinsky — APPROVED** [review]

File: src/oData/jsonToOData.ts [hunk 5]
```diff
@@ -215,18 +216,32 @@ export const convertToPineClientFilter = (
...
+
 	const { field, reverse, refScheme } = sortInfo;
 	if (!field) {
 		return null;
 	}
 	const direction = !reverse ? 'asc' : 'desc';
+	const customOrderByKey = customSort?.[field];
+	if (typeof customOrderByKey === 'string') {
+		return `${customOrderByKey} ${direction}`;
+	} else if (customOrderByKey != null && typeof customOrderByKey !== 'string') {
```
Comment (thgreasi) [inline src/oData/jsonToOData.ts:235]: I think this just needs to be v since we already checked for string:
```suggestion
	} else if (customOrderByKey != null) {
```

File: src/AutoUI/Filters/Filters.tsx [hunk 6]
```diff
@@ -23,6 +27,92 @@ interface FiltersProps<T> {
...
+	const xNoFilterSet = new Set(xNoFilter);
+	return Object.fromEntries(
+		Object.keys(properties)
+			.filter((propKey) => !xNoFilterSet.has(propKey))
+			.map((propKey) => [propKey, properties[propKey]]),
+	);
+};
+
+const removeFieldsWithNoFilter = (schema: JSONSchema): JSONSchema => {
+	const processProperties = (
```
Comment (thgreasi) [inline src/AutoUI/Filters/Filters.tsx:46]: I guess we don't really need to recreate this function every time, and given that we call it just once we could probably just inline it, right?
